src/formats/tomboy.ts

```diff
@@ -0,0 +1,134 @@
+	private coreConverter: TomboyCoreConverter;
+	private todoEnabled: boolean;
+	private keepTitleMode: KeepTitleMode;
+
+	/**
+	 * Get the default Tomboy/Gnote directory path based on the current OS
+	 */
+	private getDefaultTomboyPath(): string {
+		if (!Platform.isDesktopApp || !os || !path) {
+			return '';
+		}
+
+		try {
+			if (Platform.isMacOS) {
+				const macPath = path.join(os.homedir(), 'Library', 'Application Support', 'Tomboy');
+				return macPath;
+			} else if (Platform.isWin) {
+				const windowsPath = path.join(process.env.APPDATA || '', 'Roaming', 'Tomboy');
+				return windowsPath;
+			} else if (Platform.isLinux) {
+				// Check for both Tomboy and Gnote on Linux
+				const homeDir = os.homedir();
+				const tomboyPath = path.join(homeDir, '.local', 'share', 'tomboy');
+				const gnotePath = path.join(homeDir, '.local', 'share', 'gnote');
+				
+				// Return whichever exists, preferring Gnote since it is the newer SW
+				try {
+					if (fsPromises && fsPromises.access) {
+						// Check if gnote directory exists
+						fsPromises.access(gnotePath).then(() => {}).catch(() => {});
+						return gnotePath;
+					}
+				} catch (e) {
+					// If we can't check, default to tomboy path
+				}
+				return tomboyPath;
+			}
+		} catch (e) {
+			console.warn('Error detecting default Tomboy path:', e);
+		}
+		
+		return '';
+	}
+
+	/**
+	 * Get descriptive text for OS-specific Tomboy/Gnote locations
+	 */
+	private getOSSpecificDescription(): string {
+		if (Platform.isMacOS) {
+			return 'Tomboy notes are typically found in: ~/Library/Application Support/Tomboy';
+		} else if (Platform.isWin) {
+			return 'Tomboy notes are typically found in: %APPDATA%\\Tomboy';
+		} else if (Platform.isLinux) {
+			return 'Tomboy notes are typically found in: ~/.local/share/tomboy or ~/.local/share/gnote';
+		}
+		return 'Pick the files that you want to import.';
+	}
+
+	init() {
+		this.todoEnabled = true;
+		this.coreConverter = new TomboyCoreConverter();
+		this.keepTitleMode = 'automatic';
+
+		this.addFileChooserSetting('Tomboy/Gnote', ['note'], true, this.getOSSpecificDescription(), this.getDefaultTomboyPath());
+		this.addOutputLocationSetting('Tomboy import');
+
+		new Setting(this.modal.contentEl)
+			.setName('Convert TODO lists to checkboxes')
+			.setDesc('When enabled, lists in notes with "TODO" in the title will be converted to task lists with checkboxes. Strikethrough items will be marked as completed.')
+			.addToggle((toggle: ToggleComponent) => {
+				toggle.setValue(this.todoEnabled)
+					  .onChange((value: boolean) => this.todoEnabled = value);
+			});
+
+		new Setting(this.modal.contentEl)
+			.setName('Keep title in markdown')
+			.setDesc('Choose whether to keep the note title in the markdown content. "Automatic" keeps titles only when special characters are lost in filename conversion.')
+			.addDropdown((dropdown: DropdownComponent) => {
+				dropdown.addOption('automatic', 'Automatic')
+					.addOption('yes', 'Yes')
```

Comment:
Maybe instead of `Yes` and `No`, the description for the options could have more detail, for example something like: 

- Automatic
- Keep titles
- Filename only